waiter/src/waiter/token.clj

```diff
@@ -333,7 +377,9 @@
 
     ; Store the token
     (let [new-token-metadata (merge {"last-update-time" (.getMillis ^DateTime (clock))
-                                     "owner" owner}
+                                     "owner" owner
+                                     "root" (or (get existing-token-metadata "root")
+                                                (first waiter-hostnames))}
```

Comment:
We should make the token-root configurable in settings to avoid future potential issues with overloading the `waiter-hostnames` config value.

Reply:
Okay

Reply:
Created a new entry in `state` named `cluster-id`. However, we cannot use `:cluster-config 
 :name` from settings as we have not required it to be unique across clusters. I ended up using `:zookeeper :base-path`.